src/main/java/uk/gov/ch/oauth/tokens/SessionSignInModifier.java

```diff
@@ -0,0 +1,38 @@
+package uk.gov.ch.oauth.tokens;
+
+import java.util.Map;
+import uk.gov.companieshouse.session.Session;
+import uk.gov.companieshouse.session.SessionKeys;
+import uk.gov.companieshouse.session.model.AccessToken;
+import uk.gov.companieshouse.session.model.UserProfile;
+
+public class SessionSignInModifier {
+
+    @SuppressWarnings("unchecked")
+    // This is necessary as the original data on the session is untyped, but expected to be of the correct types
+    static Map<String, Object> updateSignIn(final Object sInf, final Object sio) {
+        final Map<String, Object> original = (Map<String, Object>) sInf;
+        final Map<String, Object> extras = (Map<String, Object>) sio;
+        original.putAll(extras);
+        return original;
+    }
+
+    public void alterSessionData(Session session, AccessToken accessToken,
+            UserProfile userProfileResponse) {
+        SignInInfoMap signInInfo = new SignInInfoMap();
+        signInInfo.setUserProfile(userProfileResponse);
+        signInInfo.setAccessToken(accessToken);
+        signInInfo.setSignedIn(true);
+
+        alterSessionData(session.getData(), signInInfo.toMap());
+    }
+
+    public void alterSessionData(Map<String, Object> currentSessionData,
+            Map<String, Object> desiredSignInInfo) {
```

Comment:
Does the method name have to be the same? It's confusing

Reply:
Yes thats what overloading does. Two methods that take different arguments but give the same result.

Reply:
I'm quite familiar with what overloading does, my point here is whether it's needed. There is no need for us to overload the method, in fact even for us to expose this method. As we would always have to set the `SignInInfoMap` data first before we go onto update the map.